v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/FailsafeModJsonToPubsubMessageTransformer.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.bigtablechangestreamstopubsub;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.api.core.ApiFuture;
+import com.google.auto.value.AutoValue;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.cloud.teleport.v2.coders.FailsafeElementCoder;
+import com.google.cloud.teleport.v2.templates.bigtablechangestreamstobigquery.model.Mod;
+import com.google.cloud.teleport.v2.templates.bigtablechangestreamstopubsub.schemautils.PubSubUtils;
+import com.google.cloud.teleport.v2.values.FailsafeElement;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.TopicName;
+import java.io.ByteArrayOutputStream;
+import java.io.Serializable;
+import java.util.concurrent.TimeUnit;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.sdk.values.TupleTagList;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Throwables;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class {@link FailsafeModJsonToPubsubMessageTransformer} provides methods that convert a JSON
+ * string wrapped in {@link FailsafeElement} to a {@link PubsubMessage}.
+ */
+public final class FailsafeModJsonToPubsubMessageTransformer {
+
+  /**
+   * Primary class for taking a {@link FailsafeElement} JSON input and converting to a {@link
+   * PubsubMessage}.
+   */
+  public static class FailsafeModJsonToPubsubMessage
+      extends PTransform<PCollection<FailsafeElement<String, String>>, PCollectionTuple> {
+
+    private final PubSubUtils pubSubUtils;
+    private static final Logger LOG =
+        LoggerFactory.getLogger(FailsafeModJsonToPubsubMessageTransformer.class);
```

Comment:
Usually we want classes to match, please change FailsafeModJsonToPubsubMessageTransformer.class to FailsafeModJsonToPubsubMessage.class